lib/rubocop/cop/sorbet/signatures/allow_incompatible_override.rb

```diff
@@ -18,43 +18,28 @@ module Sorbet
       #
       #   # good
       #   sig.override
-      class AllowIncompatibleOverride < RuboCop::Cop::Cop # rubocop:todo InternalAffairs/InheritDeprecatedCopClass
-        # @!method sig?(node)
-        def_node_search(:sig?, <<-PATTERN)
-          (
-            send
-            nil?
-            :sig
-             ...
-          )
-        PATTERN
-
-        def not_nil?(node)
-          !node.nil?
-        end
-
-        # @!method allow_incompatible?(node)
-        def_node_search(:allow_incompatible?, <<-PATTERN)
-          (pair (sym :allow_incompatible) (true))
-        PATTERN
+      class AllowIncompatibleOverride < RuboCop::Cop::Base
+        MSG = "Usage of `allow_incompatible` suggests a violation of the Liskov Substitution Principle. " \
+          "Instead, strive to write interfaces which respect subtyping principles and remove `allow_incompatible`"
+        RESTRICT_ON_SEND = [:override].freeze
 
         # @!method allow_incompatible_override?(node)
-        def_node_matcher(:allow_incompatible_override?, <<-PATTERN)
-          (
-            send
-            [#not_nil? #sig?]
+        def_node_matcher(:allow_incompatible_override?, <<~PATTERN)
+          (send
+            #sig?
             :override
-            [#not_nil? #allow_incompatible?]
-          )
+            (hash <$(pair (sym :allow_incompatible) true) ...>))
```

Comment:
Yeah, fair. I bounce around between parens on their own line, or parens on the same line, since the `ruby-parse` output spits them out on the same line, and I often copy paste that as the starting point for a node pattern.